platform/android/src/style/layers/fill_layer.cpp

```diff
@@ -40,42 +41,134 @@ namespace android {
         return jni::Object<jni::ObjectTag>(*converted);
     }
 
+
     jni::Object<jni::ObjectTag> FillLayer::getFillOpacity(jni::JNIEnv& env) {
         using namespace mbgl::android::conversion;
         Result<jni::jobject*> converted = convert<jni::jobject*>(env, layer.as<mbgl::style::FillLayer>()->FillLayer::getFillOpacity());
         return jni::Object<jni::ObjectTag>(*converted);
     }
 
+    jni::Array<jni::jlong> FillLayer::getFillOpacityTransition(jni::JNIEnv& env) {
+        using namespace mbgl::android::conversion;
+
+        mbgl::style::TransitionOptions options = layer.as<mbgl::style::FillLayer>()->FillLayer::getFillOpacityTransition();
+        jlong durations[2];
+        durations[0] = options.duration.value_or(mbgl::Duration::zero()).count();
```

Comment:
Could we just return the actual object in a typed manner. Don't think there is a noticeable overhead in creating an object instead of an array and filling it up and then creating the object from that jvm side.